lib/Monads/wp/WPC.thy

```diff
@@ -7,61 +8,64 @@
 theory WPC
 imports "WP_Pre"
 keywords "wpc_setup" :: thy_decl
-
 begin
 
-definition
-  wpc_helper :: "(('a \<Rightarrow> bool) \<times> 'b set)
-                 \<Rightarrow> (('a \<Rightarrow> bool) \<times> 'b set) \<Rightarrow> bool \<Rightarrow> bool" where
- "wpc_helper \<equiv> \<lambda>(P, P') (Q, Q') R. ((\<forall>s. P s \<longrightarrow> Q s) \<and> P' \<subseteq> Q') \<longrightarrow> R"
+(* Case splitting method producing independent guards (preconditions) for each case in a
+   datatype case split. The current setup can handle judgements such as valid, corres, or ccorres
+   with up to two independent predicate guards and one independent set typ guard. Unneeded guards
+   can be ignored in setup.
+
+   The helper predicate unifies the treatment of guards in the proof method. The P guards will be
+   transformed into Q guards in each branch of the case. The R is the judgement (valid, corres, etc).
+
+   The helper predicate encodes that the judgement supports a standard guard weakening rule.
+   From that rules for conjunction-lifting and forall-lifting follow below. These are then used
```

Comment:
suggestion: `. From that` -> `, from which` (unless you mean something else here)